assessments/projects/operator-framework/self-assessment.md

```diff
@@ -0,0 +1,338 @@
+
+- The primary application of this framework is often the creation of **custom resources**. The necessity of this feature is evidenced by the following examples:
+    - The Prometheus Operator defines eight CRDs–for example, Prometheus, Alertmanager, and ServiceMonitor among others–which allow users to specify the desired monitoring stack state.
+  - The etcd Operator on the other hand creates three CRDs–EtcdBackup, EtcdRestore, and EtcdCluster–to allow users to specify the desired state (size, version, backup policies and restore source) of the etcd clusters.
+
+-   The Operator Framework also provides libraries and tools to streamline the implementation of the **operator's reconciliation logic**, i.e., define custom controllers for operands. This is evidenced by the etcd operator for example, which defines custom controllers for each of it's three CRDs.
+
+- The **Operator Lifecycle Manager (OLM)** provides the means to bundle, distribute, and manage the operator and its dependencies. The likely use case for this is very intuitive:
+    - An update of the etcd version is rolled out.
+    - The OLM checks the CatalogSource and sees that a new ClusterServiceVersion (CSV) is available compared to the current installation.
+    - The OLM schedules an upgrade in a way that does not cause data loss, according to the reconciliation logic provided by the etcd maintainers.
+
+
+```
+Dame, M. (2022). The Kubernetes Operator Framework Book. Packt Publishing. (Chapter 10: Case Study for Optional Operator - Prometheus Operator; Chapter 11: Case Study for Core Operator – Etcd Operator)
+```
+
+### Related Projects
+
+- **Kubebuilder** is an SDK for creating Kubernetes APIs using CRDs and controllers, it is based on the controller-runtime library also used by the Operator SDK. The primary difference is that while the Kubebuilder focuses on building APIs and controllers, the Operator Framework provides additional tools for managing operator lifecycle and distributions. The Github repository is under active maintenance by the community.
+
+- **Metacontroller** is a similar framework that enables users to build custom controllers in any programming language that is supported by the Kubernetes pods. This is because it does not use CRDs or controller-runtime, it rather relies on webhooks and custom resources to communicate with the API server. Similar to Kubebuilder, it does not provide resources for operator lifecycle management and distributions; and, it is not as actively maintained as the Operator Framework.
+
+- **Crossplane** is more of a cloud focused tool, i.e., it manages connecting, consuming and provisioning cloud resources using CRDs and controllers. It also provides a package manager and container registry for the installation and distribution of controllers. The Github repository for Crossplane is under actively maintained.
+
+
+
+### Threat Modeling With STRIDE
```

Comment:
Are these mitigations recommendations to the team? Are they not implemented in the current state? If yes, in your experience in formulating this self-assessment, can you also please record all the recommendations to the team under  recommendation subsection toward the end.

Reply:
When creating them we utilized the STRIDE Threat Modeling Method to consider the potential security issues at risk. We assumed that the development team Operator Framework had already accounted for some if not most, but did not want to make any declarative statements until we received feedback from the project maintainers.

We are currently in contact with them and are waiting to receive confirmation as to which of these mitigation strategies are in place or not.

Reply:
FYI: We don't need to be overly concerned about recommendations to the project.  That usually comes up in the joint assessment (and this is only the first part).  The real focus here is just capturing the project's current state.